homeassistant/components/flume/binary_sensor.py

```diff
@@ -0,0 +1,159 @@
+    FlumeBinarySensorEntityDescription(
+        key="leak",
+        name="Leak detected",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        event_rule=NOTIFICATION_LEAK_DETECTED,
+        icon="mdi:pipe-leak",
+    ),
+    FlumeBinarySensorEntityDescription(
+        key="flow",
+        name="High flow",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        event_rule=NOTIFICATION_HIGH_FLOW,
+        icon="mdi:waves",
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up a Flume binary sensor.."""
+
+    flume_domain_data = hass.data[DOMAIN][config_entry.entry_id]
+
+    flume_auth = flume_domain_data[FLUME_AUTH]
+    flume_devices = flume_domain_data[FLUME_DEVICES]
+
+    flume_entity_list: list[
+        FlumeNotificationBinarySensor | FlumeConnectionBinarySensor
+    ] = []
+
+    connection_coordinator = FlumeDeviceConnectionUpdateCoordinator(
+        hass=hass, flume_devices=flume_devices
+    )
+    notification_coordinator = FlumeNotificationDataUpdateCoordinator(
+        hass=hass, auth=flume_auth
+    )
+
+    for device in flume_devices.device_list:
+        device_id = device[KEY_DEVICE_ID]
+        device_location_name = device[KEY_DEVICE_LOCATION][KEY_DEVICE_LOCATION_NAME]
+
+        connection_sensor = FlumeConnectionBinarySensor(
+            coordinator=connection_coordinator,
+            description=BinarySensorEntityDescription(
+                name="Connected",
+                key="connected",
+            ),
+            device_id=device_id,
+            location_name=device_location_name,
+            is_bridge=(device[KEY_DEVICE_TYPE] is FLUME_TYPE_BRIDGE),
+        )
+
+        flume_entity_list.append(connection_sensor)
+
+        if device[KEY_DEVICE_TYPE] != FLUME_TYPE_SENSOR:
+            continue
+
+        # Build notification sensors
+        flume_entity_list.extend(
+            [
+                FlumeNotificationBinarySensor(
+                    coordinator=notification_coordinator,
+                    description=description,
+                    device_id=device_id,
+                    location_name=device_location_name,
+                )
+                for description in FLUME_BINARY_NOTIFICATION_SENSORS
+            ]
+        )
+
+    if flume_entity_list:
+        async_add_entities(flume_entity_list)
+
+
+class FlumeNotificationBinarySensor(FlumeEntity, BinarySensorEntity):
+    """Binary sensor class."""
+
+    entity_description: FlumeBinarySensorEntityDescription
+    coordinator: FlumeNotificationDataUpdateCoordinator
+
+    @property
+    def is_on(self) -> bool:
+        """Return on state."""
+
+        if notifications := self.coordinator.active_notifications_by_device.get(
+            self.device_id
+        ):
+            return self.entity_description.event_rule in notifications
+
+        return False
```

Comment:
```suggestion
        return bool(
            (
                notifications := self.coordinator.active_notifications_by_device.get(
                    self.device_id
                )
            )
            and self.entity_description.event_rule in notifications
        )
```

This could be written as a single expression although its not much better since the long properties chain makes the code less compat